gsy_e_sdk/setups/Grid_Operator_API_scripts/redis_aziiz_strategy.py

```diff
@@ -0,0 +1,138 @@
+        self.import_kwh = {}
+        self.export_kwh = {}
+        self.balance_hist = []
+        self.balance = {}
+        self.dso_stats_response = {}
+
+    def on_market_cycle(self, market_info):
+        current_market_fee = {}
+        for area_uuid, area_dict in self.latest_grid_tree_flat.items():
+            if area_dict["area_name"] in MARKET_NAMES:
+                self.add_to_batch_commands.last_market_dso_stats(area_uuid)
+                current_market_fee[area_dict["area_name"]] = area_dict[
+                    "current_market_fee"
+                ]
+        self.calculate_import_export_balance()
+        next_market_fee = self.set_new_market_fee()
+        log_grid_fees_information(MARKET_NAMES, current_market_fee, next_market_fee)
+
+    def calculate_import_export_balance(self):
+        """Calculate the balance (import - export) for each market."""
+        self.dso_stats_response = self.execute_batch_commands()
+        for market_event in self.dso_stats_response["responses"].items():
+            self.import_kwh[market_event[1][0]["name"]] = market_event[1][0][
+                "market_stats"
+            ]["area_throughput"]["import"]
+            self.export_kwh[market_event[1][0]["name"]] = market_event[1][0][
+                "market_stats"
+            ]["area_throughput"]["export"]
+            self.balance[market_event[1][0]["name"]] = (
+                self.import_kwh[market_event[1][0]["name"]]
+                - self.export_kwh[market_event[1][0]["name"]]
+            )
+        self.balance_hist.append(self.balance)
+
+    def set_new_market_fee(self):
+        """Return the market fees for each market for the next time slot."""
+        next_market_fee = {}
+        for area_uuid, area_dict in self.latest_grid_tree_flat.items():
+            if area_dict["area_name"] in MARKET_NAMES:
+                if MOVING_AVERAGE_PEAK:
+                    fees = []
+                    # pylint: disable=unused-variable
+                    for j,k in enumerate(self.balance_hist):
+                        fees.append(k[area_dict["area_name"]])
+                    max_ext_energy_kwh = abs(
+                        sum(fees[-LOOK_BACK:]) / len(fees[-LOOK_BACK:])
+                    )
+                else:
+                    max_ext_energy_kwh = max(
+                        self.import_kwh[area_dict["area_name"]],
+                        self.export_kwh[area_dict["area_name"]],
+                    )
+
+                individual_market_fees = fee_strategy[area_dict["area_name"]]
+                # pylint: disable=unused-variable
+                for j, k in enumerate(individual_market_fees):
+                    if (
+                        max_ext_energy_kwh
+                        <= k["Import / Export threshold"]
+                    ):
+                        next_market_fee[
+                            area_dict["area_name"]
+                        ] = k["Grid fee"]
+                        break
```

Comment:
I think it fits in one line, right?
```suggestion
                    if (max_ext_energy_kwh <= market_fee["Import / Export threshold"]):
                        next_market_fee[area_dict["area_name"]] = market_fee["Grid fee"]
                        break
```

Reply:
Also not addressed yet